app/avatar/models.py

```diff
@@ -238,60 +177,28 @@ def create_from_config(self):
             avatar.save(tmp.name)
             with open(tmp.name) as file:
-                if self.profile_set.exists():
-                    profile = self.profile_set.last()
+                if self.profile:
+                    profile = self.profile
 
                 svg_name = profile.handle if profile and profile.handle else token_hex(8)
-                self.svg.save(f"{svg_name}.svg", File(file), save=True)
-
-    def convert_field(self, source, target='custom_avatar_png', input_fmt='svg', output_fmt='png', force_save=False):
-        """Handle converting from the source field to the target based on format."""
-        try:
-            # Convert the provided source to the specified output and store in BytesIO.
-            if output_fmt == 'svg':
-                tmpfile_io = convert_wand(source, input_fmt=input_fmt, output_fmt=output_fmt)
-            else:
-                tmpfile_io = convert_img(source, input_fmt=input_fmt, output_fmt=output_fmt)
-            if self.profile_set.exists():
-                png_name = self.profile_set.last().handle
-            else:
-                png_name = token_hex(8)
+                self.svg.save(f"{svg_name}.svg", File(file), save=False)
 
-            if tmpfile_io:
-                custom_avatar = ContentFile(tmpfile_io.getvalue())
-                custom_avatar.name = f'{png_name}.{output_fmt}'
-                setattr(self, target, custom_avatar)
-                if force_save:
-                    self.save()
-                return True
-        except Exception as e:
-            logger.error('Error: (%s) - Avatar PK: (%s)', str(e), self.id)
-        return False
+    def to_dict(self):
+        return self.config
 
-    def convert_custom_svg(self, force_save=False):
-        """Handle converting the custom Avatar SVG to PNG."""
-        try:
-            if not self.svg:
-                self.create_from_config()
-
-            converted = self.convert_field(
-                self.svg, 'custom_avatar_png', input_fmt='svg', output_fmt='png', force_save=force_save
-            )
-            if not converted:
-                raise AvatarConversionError('Avatar conversion error while converting SVG!')
-        except AvatarConversionError as e:
-            logger.error('Error: (%s) - Avatar PK: (%s)', str(e), self.id)
 
-    def convert_github_png(self, force_save=False):
-        """Handle converting the Github Avatar PNG to SVG."""
-        try:
-            if not self.png:
-                self.pull_github_avatar()
-
-            converted = self.convert_field(
-                self.png, 'github_svg', input_fmt='png', output_fmt='svg', force_save=force_save
-            )
-            if not converted:
-                raise AvatarConversionError('Avatar conversion error while converting SVG!')
-        except AvatarConversionError as e:
-            logger.error('Error: (%s) - Avatar PK: (%s)', str(e), self.id)
+class SocialAvatar(BaseAvatar):
+
```

Comment:
any social avatar specific fields ? 
i was thinking about `platform` but now there is only github so no need to add it atm